gobblin-runtime/src/main/java/gobblin/runtime/mapreduce/MRJobLauncher.java

```diff
@@ -249,37 +248,37 @@ protected void executeCancellation() {
   /**
    * Add dependent jars and files.
    */
-  private void addDependencies()
+  private void addDependencies(Configuration conf)
       throws IOException {
     TimingEvent distributedCacheSetupTimer =
         this.eventSubmitter.getTimingEvent(TimingEvent.RunJobTimings.MR_DISTRIBUTED_CACHE_SETUP);
 
-    Path jarFileDir = new Path(this.mrJobDir, JARS_DIR_NAME);
+    Path jarFileDir = this.jarsDir;
```

Comment:
Line 362 in `MRJobLauncher` is this: `       DistributedCache.addFileToClassPath(destJarFile, this.conf, this.fs);
`, so the jars get added individually to the distributed classpath, the directory is just a container for all jars.